### PR TITLE
issue: 3875902: [TFS plugin] Improve the CSV processing time for the telemetry data

### PR DESCRIPTION
**Pull request by ananalaghbar**
## What ?
Implement a new clean function for only converting the CSV telemetry data to JSON (list of ports) + skip the empty values + skip the disabled counters  + convert the numeric values from string to float or integer + append the meta fields **WITHOUT** calculating the delta

## Why ?
The updated function simplifies the processing of CSV data, reducing the time to handle 200K ports and 31 counters from about 9 seconds to 2.5 seconds.

### CODE REVIEW

File: plugins/fluentd_telemetry_plugin/conf/fluentd.conf [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+  @type forward
+#  @type http
   bind 0.0.0.0
   port 24225 
 </source>
 <match UFM_Telemetry_Streaming**>
   @type stdout
 #  @type file
-#  path /tmp/fluentd-stremaing2/data.log
+#  path /tmp/fluentd-stremaing/data.log
```
Comment (mike-dubman) [inline plugins/fluentd_telemetry_plugin/conf/fluentd.conf:10]: typo: s/stremaing/streaming

File: plugins/fluentd_telemetry_plugin/src/streamer.py [hunk 2]
```diff
@@ -385,63 +381,92 @@ def _get_saved_streaming_attributes(self):
     def update_saved_streaming_attributes(self, attributes):
         Utils.write_json_to_file(self.streaming_attributes_file, attributes)
 
-    def _parse_telemetry_csv_metrics_to_json(self, data, line_separator="\n", attrs_separator=","):
+    def _get_filtered_counters(self, counters):
```
Comment (haithamwj) [inline plugins/fluentd_telemetry_plugin/src/streamer.py:384]: let us document the function and what it does

**Review by haithamwj — APPROVED** [review]

File: plugins/fluentd_telemetry_plugin/src/streamer.py [hunk 3]
```diff
@@ -385,63 +381,92 @@ def _get_saved_streaming_attributes(self):
...
+        keys_length = len(counters)
+        modified_keys = {}
+        for i in range(keys_length):
+            key = counters[i]
+            attr_obj = self.streaming_attributes.get(key, {'enabled': True})
```
Comment (Abeer-Moghrabi) [inline plugins/fluentd_telemetry_plugin/src/streamer.py:389]: The default value should be removed

**Review by Abeer-Moghrabi — CHANGES_REQUESTED** [review]
please see my comment

**Review by Abeer-Moghrabi — APPROVED** [review]

File: plugins/fluentd_telemetry_plugin/src/streamer.py (hunk 3 above)
Comment (ananalaghbar) [reply plugins/fluentd_telemetry_plugin/src/streamer.py]: done

File: plugins/fluentd_telemetry_plugin/src/streamer.py (hunk 2 above)
Comment (ananalaghbar) [reply plugins/fluentd_telemetry_plugin/src/streamer.py:388]: done

File: plugins/fluentd_telemetry_plugin/src/streamer.py [hunk 4]
```diff
@@ -503,6 +570,7 @@ def _stream_data_to_fluentd(self, data_to_stream, fluentd_msg_tag=''):
             }
 
             if self.compressed_streaming_flag:
+                plugin_name = 'HTTP'
```
[the remark this answers is not in the document]
Comment (ananalaghbar) [reply plugins/fluentd_telemetry_plugin/src/streamer.py]: done

File: plugins/fluentd_telemetry_plugin/src/streamer.py [hunk 5]
```diff
@@ -385,63 +381,134 @@ def _get_saved_streaming_attributes(self):
...
+
+        is_meta_fields_available = len(self.meta_fields[0]) or len(self.meta_fields[1])
+        modified_keys = self._get_filtered_counters(keys)
+        available_keys_indices = modified_keys.keys()
+
+        for row in rows[1:-1]:
+            values = row.split(attrs_separator)
+            port_record = {}
+            for i in available_keys_indices:
+                value = values[i]
```
[the remark this answers is not in the document]
Comment (ananalaghbar) [reply plugins/fluentd_telemetry_plugin/src/streamer.py:502]: The `available_keys_indices` correspond to the keys in `modified_keys`, which is derived from the headers matching the row values in length, eliminating the risk of out-of-range exceptions.

**Review by TalLerner — APPROVED** [review]

**Pull request merged**